hbase-server/src/main/java/org/apache/hadoop/hbase/master/MasterRpcServices.java

```diff
@@ -3321,4 +3325,33 @@ public UpdateRSGroupConfigResponse updateRSGroupConfig(RpcController controller,
     }
     return builder.build();
   }
+
+  @Override
+  public RegionServerReportResponse replicationServerReport(RpcController controller,
+      RegionServerReportRequest request) throws ServiceException {
+    try {
+      master.checkServiceStarted();
+      int versionNumber = 0;
+      String version = "0.0.0";
+      VersionInfo versionInfo = VersionInfoUtil.getCurrentClientVersionInfo();
+      if (versionInfo != null) {
+        version = versionInfo.getVersion();
```

Comment:
The difference of version and versionNumber is?

Reply:
`versionNumber` is packed by number part of `version`, can't display string suffixes like "-SNAPSHOT".